modules/ROOT/pages/oauth/oauth2-provider-documentation-reference.adoc

```diff
@@ -241,171 +322,222 @@ A new refresh token will be generated every time a refresh token request is exec
 
-==== Parameters
-[cols=".^20%,.^20%,.^35%,.^20%,^.^5%", options="header"]
-|===
-| Name | Type | Description | Default Value | Required
-| Config | String | The name of a globally defined OAuth Provider configuration to use for token validation. | {nbsp}| *x*{nbsp}
-| Client Id | String | The Id to assign to the created client | {nbsp} | *x*{nbsp}
-| Client Name | String | The client friendly name | {nbsp} | {nbsp}
-| Principal | String | For some security providers, the clientId can't be used for the client username. In those cases, the client's principal is used for authentication. | {nbsp} | {nbsp}
-| Description | String | A short description of a client | {nbsp} | {nbsp}
-| Type | Client Type | The type of the client. Allowed values are PUBLIC (Clients incapable of maintaining the confidentiality of their credentials) or CONFIDENTIAL (Clients capable of maintaining the confidentiality of their credentials). | PUBLIC | {nbsp}
-| Secret | String | The client's secret (password) used for authentication. | {nbsp} | Only if the client type is CONFIDENTIAL
-| Redirect Uris | Expression | An expression that resolves to a list of redirect uris used for when the client makes requests to the OAuth Provider. | Empty List | {nbsp}
-| Authorized Grant Types | Expression | An expression that resolves to a list of the authorized grant types that the client can use to request a token. Valid values are: AUTHORIZATION_CODE,REFRESH_TOKEN, TOKEN, PASSWORD, CLIENT_CREDENTIALS. | Empty List | {nbsp}
-| Scopes | Expression |  An expression that resolves to a list of supported scopes by the client. If none provided, the default scopes of the <<general-configuration>> will be used. | Empty List | {nbsp}
-| Fail if Present | Boolean | Defines what to do if the a client with the same ID is already registered. If true, an error will be raised. Otherwise, the client will be updated. | false | {nbsp}
-|===
+Apart from that, the provider defines an operation: <<validateToken>> that can check if a token is authorized. That operation can be added anywhere in a flow to control it's execution. If the token is indeed authorized,
+the flow will keep executing, setting token information in the payload, otherwise, a TOKEN_UNAUTHORIZED error will be raised. You need to add that operation in parts of the application that need token authorization.
 
-==== Raises
-* OAUTH2-PROVIDER:CLIENT_ALREADY_EXISTS(OAUTH_SERVER_SECURITY)
-+
-{nbsp} If a client already exists with the same client ID, and the flag: failIfPresent is set to true.
-+
-* INVALID_CONFIGURATION
-+
-{nbsp} If the provided parameters are not valid, as having an authorizationGrantType of AUTHORIZATION_CODE and no redirect URI.
+Also, since token validation is almost always going to be used together with an HTTP Listener, in case it fails, the listener's response mechanism can handle that error and properly respond to the requester. Additional logic can be added handling that type of error.
```

Comment:
```suggestion
Since token validation is almost always used together with an HTTP Listener, in case it fails, the Listener's response mechanism can handle the error and properly respond to the requester. Additional logic can be added for handling that type of error.
```